crates/compiler/solve/src/ability.rs

```diff
@@ -849,6 +859,104 @@ trait DerivableVisitor {
     }
 }
 
+struct DeriveInspect;
+impl DerivableVisitor for DeriveInspect {
+    const ABILITY: Symbol = Symbol::INSPECT_INSPECT_ABILITY;
+    const ABILITY_SLICE: SubsSlice<Symbol> = Subs::AB_INSPECT;
+
+    #[inline(always)]
+    fn is_derivable_builtin_opaque(symbol: Symbol) -> bool {
+        // TODO: Should this just be true? All values are always inspectable.
+        is_builtin_number_alias(symbol) || is_builtin_bool_alias(symbol)
+    }
+
+    #[inline(always)]
+    fn visit_recursion(_var: Variable) -> Result<Descend, NotDerivable> {
+        Ok(Descend(true))
+    }
+
+    #[inline(always)]
+    fn visit_apply(var: Variable, symbol: Symbol) -> Result<Descend, NotDerivable> {
+        if matches!(
+            symbol,
+            Symbol::LIST_LIST | Symbol::SET_SET | Symbol::DICT_DICT | Symbol::STR_STR,
+        ) {
+            Ok(Descend(true))
+        } else {
+            Err(NotDerivable {
+                var,
+                context: NotDerivableContext::NoContext,
+            })
+        }
```

Comment:
Pretty sure this should also be `true` unconditionally, or let's panic in the `else` branch.